packages/web/src/components/lineup/InfiniteCardLineup.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useCallback, useRef } from 'react'
+
+import InfiniteScroll from 'react-infinite-scroller'
+
+import LoadingSpinner from 'components/loading-spinner/LoadingSpinner'
+import { getScrollParent } from 'utils/scrollParent'
+
+import CardLineup, { CardLineupProps } from './CardLineup'
+import styles from './InfiniteCardLineup.module.css'
+
+type InfiniteLoadingProps = {
+  hasMore: boolean
+  loadMore: () => void
+}
+
+export type InfiniteCardLineupProps = CardLineupProps & InfiniteLoadingProps
+
+const InfiniteCardLineup = (props: InfiniteCardLineupProps) => {
+  const { hasMore, loadMore, ...lineupProps } = props
+  const scrollRef = useRef(null)
+
+  const getNearestScrollParent = useCallback(() => {
+    if (!scrollRef.current) {
+      return null
+    }
+    return (
+      (getScrollParent(scrollRef.current) as unknown as HTMLElement) ?? null
```

Comment:
I looked into this. `getScrollParent` returns `document.scrollElement` in some cases, which is typed as `Element`. I'm not 100% sure why. But it's a built-in type, so I don't think we could change it, other than to move this cast into `getScrollParent` itself. Thoughts on that?

Reply:
It looks like in most other places we are defining `getScrollParent` locally in the component file. I only see one other usage of `utils/scrollParent`. Maybe it would be better to define locally?